aiohttp/client_reqrep.py

```diff
@@ -82,6 +88,10 @@ def __init__(self, method, url, *,
         self.url = url.with_fragment(None)
         self.original_url = url
         self.method = method.upper()
+
+        if self.method in self.GET_METHODS and any((data, compress, chunked)):
+            raise ValueError('Method %s should not contain request-data.' % self.method)
```

Comment:
This will eventuall break requests to elastic search. Technically, you can send GET with payload and that's correct from the point of spec.

https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-body.html